guillotina/tests/test_dynamic_schema.py

```diff
@@ -491,7 +491,12 @@ def get_foo(inst):
                     }
                 }
             }))
-        assert status == 412
+        assert status == 204
```

Comment:
@vangheem  I changed this because I think that this call is correct... Can you confirm?

Reply:
hmmm, I don't think so. This is a test to specifically make sure it gives a configuration error if you provide the wrong type for a value.

Was this test always wrong maybe?

`foobar` is an int though and it's trying to patch it as a string.

Reply:
This test was passing in the past because the field type was `int` instead of `integer` and this was producing a 412 for any kind of value. If tried patching with a `5` (without quotes) and is also returning a 412.

So, after fixing the PatchField thing, guillotina started complaining that `int` is not a valid `type`.

Now, when doing the PATCH with a quoted integer the `IJSONToValue` transforms the string to integer without complaining https://github.com/plone/guillotina/blob/master/guillotina/fields/dynamic.py#L63 . I guess this is happening in other places. Is this behavior correct?

Also, I updated the tests to verify that a string like `"not an integer"` fails with 412

Reply:
cool, looks good then.